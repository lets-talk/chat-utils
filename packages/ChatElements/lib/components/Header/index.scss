@import "./config.scss";

@mixin letstalk-header {
  flex-shrink: 0;
  position: relative;
}

@mixin letstalk-title {
  font-size: $widgetBox-Header-title-fontsize;
}

@mixin letstalk-minimize-button {
  background-color: transparent;
  border: 0;
  display: block;
  position: absolute;
  left: 10px;
  top: 15px;
  width: 40px;
}

@mixin letstalk-menu-button {
  background-color: transparent;
  border: 0;
  display: block;
  position: absolute;
  right: 10px;
  top: 15px;
  width: 40px;
}

.letstalk-header {
  @include letstalk-header;
  @include themify($themes) {
    color: themed('color-text');
    background-color: themed('color-background-primary');
  }
  border-radius: 10px 10px 0 0;
  color: white;
  display: flex;
  flex-direction: column;
  text-align: $widgetBox-Header-text-align;
  padding: 15px 20px;
}

.letstalk-title {
  @include letstalk-title;
  font-size: $widgetBox-Header-title-fontsize;
  font-weight: normal;
  margin: 0 40px;
  padding: 5px 10px;
}

.letstalk-subtitle {
  font-size: $widgetBox-Header-subtitle-fontsize;
}

.letstalk-minimize-button {
  @include letstalk-minimize-button;
}

.letstalk-menu-button {
  @include letstalk-menu-button;
}
