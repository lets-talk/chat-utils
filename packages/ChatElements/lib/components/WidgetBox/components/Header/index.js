import React from 'react';
import PropTypes from 'prop-types';

import MdKeyboardArrowDown from 'react-icons/lib/md/keyboard-arrow-down';
import MdMenu from 'react-icons/lib/md/menu';
import Avatar from '../../../Avatar';

import './index.scss';

const Header = ({
  title, subtitle, avatar, toggleChat, openMenu, showMinimizeButton, showMenuButton,
}) =>
  (
    <div className="letstalk-header">
      {avatar && <Avatar src={avatar} size="medium" withStatus status="online" />}
      <h4 className="letstalk-title">{title}</h4>
      <span className="letstalk-subtitle">{subtitle}</span>
      {
        showMinimizeButton &&
        <button className="letstalk-minimize-button" onClick={toggleChat}>
          <MdKeyboardArrowDown size={20} color="white" />
        </button>
      }
      {
        showMenuButton &&
        <button className="letstalk-menu-button" onClick={openMenu}>
          <MdMenu size={20} color="white" />
        </button>
      }
    </div>
  );

Header.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  avatar: PropTypes.string,
  toggleChat: PropTypes.func,
  openMenu: PropTypes.func,
  showMinimizeButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
};

export default Header;
