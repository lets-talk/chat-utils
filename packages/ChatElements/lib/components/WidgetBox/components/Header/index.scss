@import '../../../../common/styles/variables.css';
@import "../common.scss";

.letstalk-header {
  background-color: var(--widgetBox-Header-background);
  border-radius: 10px 10px 0 0;
  color: white;
  display: flex;
  flex-direction: column;
  text-align: var(--widgetBox-Header-text-align);
  padding: 15px 20px;
}

.letstalk-title {
  font-size: var(--widgetBox-Header-title-fontsize);
  font-weight: normal;
  margin: 0 40px;
  padding: 5px 10px;
}

.letstalk-subtitle {
  font-size: var(--widgetBox-Header-subtitle-fontsize);
}

.letstalk-minimize-button {
  display: none;
}

@media screen and (max-width: 800px) {
  .letstalk-header {
    @include letstalk-header;
  }

  .letstalk-title {
    @include letstalk-title;
  }

  .letstalk-minimize-button {
    @include letstalk-minimize-button;
  }

  .letstalk-minimize {
    @include letstalk-minimize;
  }

  .letstalk-menu-button {
    @include letstalk-menu-button;
  }

  .letstalk-menu {
    @include letstalk-menu;
  }
}
