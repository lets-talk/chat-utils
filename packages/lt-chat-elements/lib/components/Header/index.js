import React from 'react';
import PropTypes from 'prop-types';

import MdKeyboardArrowDown from 'react-icons/lib/md/keyboard-arrow-down';
import styled from 'styled-components';
import { Flex, Box } from 'grid-styled';
import MdMenu from 'react-icons/lib/md/menu';
import Avatar from '../Avatar';
import { headerMenuButton, themeColor, textColor } from '../../utils/style';

const StyledHeaderContainer = styled.div`
  background-color: ${(props) => themeColor(props.theme, 'accent', 'base')};
  color: ${(props) => textColor(props.theme, 'dark', 'primary')};
  border-radius: 10px 10px 0 0;
  text-align: center;
  padding: 15px 20px;

  button {
    color: ${(props) => textColor(props.theme, 'dark', 'primary')};
    ${headerMenuButton('40px')};
  }

  h4 {
    margin: 0;
    font-size: ${(props) => props.theme.typography.classes.body.fontSize};
    line-height: ${(props) => props.theme.typography.classes.body.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  }

  span {
    color: ${(props) => textColor(props.theme, 'dark', 'secondary')};
    font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
    line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightThin};
  }
`;

const StyledAvatarContainer = styled.div`
  margin-bottom: 10px;
`;

const Header = ({
  className, title, subtitle, person, toggleChat, openMenu, showMinimizeButton, showMenuButton,
}) => {
  const { avatar = false, status = '' } = person;
  return (
    <StyledHeaderContainer className={className}>
      <Flex alignItems="center">
        <Box flex="0 0 auto" mb="auto" width="40px">
          {showMenuButton &&
            <button onClick={openMenu}>
              <MdMenu size={20} />
            </button>
          }
        </Box>

        <Box flex="1 1 auto">
          {avatar &&
            <StyledAvatarContainer>
              <Avatar
                src={avatar}
                size="medium"
                withStatus={status.length > 0}
                status={status}
              />
            </StyledAvatarContainer>
          }
          {title && <h4>{title}</h4>}
          {subtitle && <span>{subtitle}</span>}
        </Box>

        <Box flex="0 0 auto" mb="auto" width="40px">
          {showMinimizeButton &&
            <button onClick={toggleChat}>
              <MdKeyboardArrowDown size={20} />
            </button>
          }
        </Box>
      </Flex>
    </StyledHeaderContainer>
  );
};

Header.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  person: PropTypes.shape({
    avatar: PropTypes.string,
    status: PropTypes.string,
    email: PropTypes.string,
    type: PropTypes.string,
  }),
  toggleChat: PropTypes.func,
  openMenu: PropTypes.func,
  showMinimizeButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
};

Header.defaultProps = {
  className: 'LT-Header-Container',
  title: '',
  subtitle: '',
  person: {},
  toggleChat: null,
  openMenu: null,
  showMinimizeButton: true,
  showMenuButton: true,
};

Header.displayName = 'Header';

export default Header;
