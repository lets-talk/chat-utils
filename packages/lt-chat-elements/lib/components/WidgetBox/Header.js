import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import { headerMenuButton, themeColor, textColor } from '../../utils/style';

const StyledHeaderContainer = styled.div`
  flex: 0 0 auto;
  flex-basis: ${(props) => `${props.height}px`};

  background-color: ${(props) => themeColor(props.theme, 'accent', 'base')};
  color: ${(props) => textColor(props.theme, 'dark', 'primary')};
  border-radius: 10px 10px 0 0;
  text-align: center;
  padding: 15px 20px;

  button {
    color: ${(props) => textColor(props.theme, 'dark', 'primary')};
    ${headerMenuButton('40px')};
  }

  h4 {
    margin: 0;
    font-size: ${(props) => props.theme.typography.classes.body.fontSize};
    line-height: ${(props) => props.theme.typography.classes.body.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  }

  span {
    color: ${(props) => textColor(props.theme, 'dark', 'secondary')};
    font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
    line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightThin};
  }
`;
StyledHeaderContainer.displayName = 'StyledHeaderContainer';

const Header = (props) => {
  const { children, className, height } = props;
  return (
    <StyledHeaderContainer className={className} height={height}>
      {children}
    </StyledHeaderContainer>
  );
};

Header.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  height: PropTypes.number,
};

Header.defaultProps = {
  className: 'LT-Header-Container',
  height: 50,
};

Header.displayName = 'Header';

export default Header;
