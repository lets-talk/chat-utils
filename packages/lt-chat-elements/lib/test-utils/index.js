// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
