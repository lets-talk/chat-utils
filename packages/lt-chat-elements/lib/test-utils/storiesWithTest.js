import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
