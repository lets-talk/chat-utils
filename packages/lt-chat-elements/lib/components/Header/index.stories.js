import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, select, text, object } from '@storybook/addon-knobs';

// Component to show on storybook
import Header from '.';
import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('Header', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('Header'));

const leftButtons = [
  {
    onClick: () => action('onLeftButtonClick'),
    icon: <i>Icon Left</i>,
  },
];

const rightButtons = [
  {
    onClick: () => action('onRightButtonClick'),
    icon: <b>Any node</b>,
  },
];

const agent = {
  avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
  email: '',
  type: 'Agent',
  status: 'online',
};

stories.addWithInfo(
  'Header',
  'This is the basic usage with the Header',
  () => (
    <div
      style={{
        position: 'fixed', bottom: 0, right: '10px', width: 'calc(100% - 20px)',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <Header
          title={text('Title', 'Ayuda en LÃ­nea')}
          subtitle={text('Subtitle', 'Cambio de clave')}
          toggleChat={action('toggleChat')}
          openMenu={action('openMenu')}
          showMenuButton={boolean('Show Menu Button', false)}
          showMinimizeButton={boolean('Show Minimize Button', false)}
          person={object('Person', agent)}
          leftButtons={leftButtons}
          rightButtons={rightButtons}
        />
      </WrapWithTheme>
    </div>
  )
);
