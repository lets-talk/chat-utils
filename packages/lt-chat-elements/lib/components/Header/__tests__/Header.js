import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
import Header from '../index';

import { mountWithTheme } from '../../../test-utils';

describe('Header component', () => {
  it('should render with minimal props', () => {
    const component = shallow(<Header />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data', () => {
    const props = {
      title: 'Header Title',
      subtitle: 'Header subtitle',
      avatar: '',
      avatarStatus: 'online',
      toggleChat: () => null,
      openMenu: () => null,
      showMinimizeButton: true,
      showMenuButton: true,
    };

    const component = shallow(<Header {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Header with openMenu and toggleChat functions handlers', () => {
    it('Test toggleChat function handler is called when click on proper button', () => {
      const mockToggleChat = jest.fn();
      const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

      const props = {
        title: 'Header Title',
        subtitle: 'Header subtitle',
        avatar: '',
        avatarStatus: 'online',
        toggleChat: mockToggleChat,
        showMinimizeButton: true,
        showMenuButton: true,
      };

      const wrapper = mountWithTheme(<Header toggleChat={mockToggleChat} {...props} />);
      wrapper.find('button').last().simulate('click', fakeEventObject);
      expect(mockToggleChat.mock.calls.length).toEqual(1);
    });

    it('Test openMenu function handler is being called when clicked on proper button', () => {
      const mockOpenMenu = jest.fn();
      const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

      const props = {
        title: 'Header Title',
        subtitle: 'Header subtitle',
        avatar: '',
        avatarStatus: 'online',
        openMenu: mockOpenMenu,
        showMinimizeButton: true,
        showMenuButton: true,
      };

      const wrapper = mountWithTheme(<Header openMenu={mockOpenMenu} {...props} />);
      wrapper.find('button').first().simulate('click', fakeEventObject);
      expect(mockOpenMenu.mock.calls.length).toEqual(1);
    });
  });
});
