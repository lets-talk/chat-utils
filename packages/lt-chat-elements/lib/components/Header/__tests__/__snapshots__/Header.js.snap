// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render with data 1`] = `
<styled.div
  className="LT-Header-Container"
>
  <Flex
    alignItems="center"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      flex="0 0 auto"
      mb="auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width="40px"
    >
      <button
        onClick={[Function]}
      >
        <MdMenu
          size={20}
        />
      </button>
    </Box>
    <Box
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <h4>
        Header Title
      </h4>
      <span>
        Header subtitle
      </span>
    </Box>
    <Box
      flex="0 0 auto"
      mb="auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width="40px"
    >
      <button
        onClick={[Function]}
      >
        <MdKeyboardArrowDown
          size={20}
        />
      </button>
    </Box>
  </Flex>
</styled.div>
`;

exports[`Header component should render with minimal props 1`] = `
<styled.div
  className="LT-Header-Container"
>
  <Flex
    alignItems="center"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      flex="0 0 auto"
      mb="auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width="40px"
    >
      <button
        onClick={null}
      >
        <MdMenu
          size={20}
        />
      </button>
    </Box>
    <Box
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    />
    <Box
      flex="0 0 auto"
      mb="auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width="40px"
    >
      <button
        onClick={null}
      >
        <MdKeyboardArrowDown
          size={20}
        />
      </button>
    </Box>
  </Flex>
</styled.div>
`;
